src/backend/brave_search/brave_client.py

```diff
@@ -0,0 +1,302 @@
+import os
+from enum import StrEnum
+from typing import Dict, Optional
+
+import httpx
+import requests
+from pydantic import BaseModel
+from tenacity import AsyncRetrying, stop_after_attempt, wait_fixed
+
+
+class SafeSearchTypes(StrEnum):
+    OFF = "off"
+    MODERATE = "moderate"
+    STRICT = "strict"
+
+
+class WebSearchResponse(BaseModel):
+    """
+    Web search response model.
+    """
+
+    type: str = "search"
+    discussions: Dict = {}
+    faq: Dict = {}
+    infobox: Dict = {}
+    locations: Dict = {}
+    mixed: Dict = {}
+    news: Dict = {}
+    query: Dict = {}
+    videos: Dict = {}
+    web: Dict = {}
+    summarizer: Dict = {}
+
+
+class BraveApiException(Exception):
+    """
+    Custom exception for Brave API errors.
+    """
+    pass
+
+
+class BraveClient:
+    BASE_URL = "https://api.search.brave.com/res/v1/"
+    WEB_SEARCH_API_ENDPOINT = "web/search"
+    MAX_RESULTS = 20
+    MAX_OFFSET = 9
+    WAIT_RETRY_SECONDS = 2
+    RETRY_ATTEMPTS = 2
+
+    def __init__(self, api_key: Optional[str] = None) -> None:
```

Comment:
Rather than os.environ, can we use `Settings()` here to fetch the API key

Reply:
I used os.environ in the Brave client as a fallback. The Settings are used in the tool to get the API key.